packages/polling/src/ratelimiter.ts

```diff
@@ -133,27 +159,33 @@ export class Throttler<T = any, U = any> extends RateLimiter<T, U> {
    * #### Notes
    * The `edge` defaults to `leading`; the `limit` defaults to `500`.
    */
-  constructor(fn: () => T | Promise<T>, options?: Throttler.IOptions | number) {
+  constructor(
+    fn: (...args: V) => T | Promise<T>,
+    options?: Throttler.IOptions | number
+  ) {
     super(fn, typeof options === 'number' ? options : options && options.limit);
-    let edge: 'leading' | 'trailing' = 'leading';
-    if (typeof options !== 'number') {
-      options = options || {};
-      edge = 'edge' in options ? options.edge! : edge;
+    if (typeof options !== 'number' && options && options.edge === 'trailing') {
```

Comment:
Small compactor suggestion

```suggestion
    if (typeof options !== 'number' && options?.edge === 'trailing') {
```

Reply:
@fcollonval, I wish I could do that here. The version of TypeScript that Lumino uses doesn't have that feature. It hurts every time it comes up 🥲

Reply:
<img width="402" alt="Screen Shot 2022-07-21 at 11 25 32" src="https://user-images.githubusercontent.com/159529/180192333-5770780d-d248-4793-afdf-ddf95b406131.png">

Reply:
Lumino uses TypeScript 3.6 and the chaining operator (`?.`) was introduced in 3.7. I don't know if we still need to be on this old version of the language though. Perhaps @blink1073 or @jasongrout remember the context and reason.

Reply:
@fcollonval reminded me of this: https://github.com/jupyterlab/lumino/pull/175

Sorry for the noise, Jason and Steve!